addons/auth_passkeys/static/src/create_passkey_button.js

```diff
@@ -0,0 +1,71 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+
+import { Component } from "@odoo/owl";
+
+export class CreatePasskeyButton extends Component {
+
+    setup() {
+        this.orm = useService("orm")
+        this.notification = useService("notification")
+        this.dialogService = useService("dialog")
+        this.actionService = useService("action")
+        this.title = _t('Create a new Passkey')
+    }
+
+    async createNewPasskey() {
+        const name = this.props.record.data.name
+        if(name.length == 0) {
+            this.notification.add(_t("The Passkey needs a name"), {
+                title: _t("Error"),
+                type: "danger",
+            })
+            return
+        }
+        try {
+            const registration = await this.register()
+            this.orm.create("auth.passkeys.key", [{
```

Comment:
We should go through a wizard, do the same thing done as for API keys
https://github.com/odoo/odoo/blob/07f6d71e4dd3fac7b4cacc545819430aa48823cd/addons/portal/static/src/js/portal_security.js#L41

API keys relies on Javascript only for portals. For regular employees users, it uses the regular wizard framework, without any additional Javascript.

In your case, you need Javascript calls to communicate with the browser to negotiate the passkey creation, hence I understand a simple wizard is not enough.
However, instead of doing a full Javascript component for in the end creating dialogs which are very similar to a wizard, I wonder how much work it would be to override in Javascript the wizard behavior, to override the primary button behavior to do these Javascript calls upon a click on that button. I might be in the end less Javascript code than what you have currently.

Because, in the end, there is no so much difference between your passkey creation flow and the API key creation flow, at least for the user:
- A button "Add a API key / Passkey" in the user profile
- Opening a dialog asking you to name your API Key / Passkey
- The API key flow has a confirmation dialog telling the key was created, which you do not have currently for the passkey, but which you should have in my opinion. Currently your passkey creation flow closes abruptly, without letting the user knowing the passkey was created successfully.